testing/sliding-sync-integration-test/src/lib.rs

```diff
@@ -1335,14 +1218,7 @@ mod tests {
 
         // let's get that first entry
 
-        let Some(RoomListEntry::Invalidated(room_id)) = view
-            .rooms_list
-            .lock_ref()
-            .iter()
-            .next()
-            .map(Clone::clone) else {
-            panic!("2nd room has moved? how?");
-        };
+        let room_id = assert_matches!(view.rooms_list().get(0), Some(&RoomListEntry::Invalidated(ref room_id)) => room_id.clone());
```

Comment:
Yeah, but the entire `RoomListEntry` is cloned (because I'm written a new impl `From<&RoomListEntry> for RoomListEntry`), to only fetch the `room_id`. I thought it would save some bits when done like this. Thoughts?

Reply:
Anyway, Clippy is complaining about it. I've removed `&` and `ref`. That's just tests after all :-p.

Reply:
Both ways to write it are exactly equivalent.